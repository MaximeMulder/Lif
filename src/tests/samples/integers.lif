print(0);
print(1);
print(227);

print(2 == 2);
print(2 != 2);
print(2 <= 2);
print(2 >= 2);

print(2 < 3);
print(2 > 3);
print(2 <= 3);
print(2 >= 3);
print(2 == 3);
print(2 != 3);

print(3 < 2);
print(3 > 2);
print(3 <= 2);
print(3 >= 2);
print(3 == 2);
print(3 != 2);

print(+0);
print(-0);
print(-1);
print(--1);

print(0016);
print(0b10000);
print(0o20);
print(0x10);

print(1_000);
print(0b1000_1000);

print(12 + 2);
print(12 - 2);
print(12 * 2);
print(12 / 2);
print(12 % 2);

print(~21);

print(21 & 7);
print(21 | 7);
print(21 ^ 7);

print(17 >> 1);
print(17 << 1);

print(5.6);
print(-5.6);

print(if true { 1 } else { 0 });
print(if false { 1 } else { 0 });
