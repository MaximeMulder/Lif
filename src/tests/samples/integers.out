0
1
227
true
false
true
true
true
false
true
false
false
true
false
true
false
true
false
true
0
0
-1
1
16
16
16
16
1000
136
14
10
24
6
0
-22
5
23
18
8
34
5.6
-5.6
1
0
